### PR TITLE
docs: introduce new documentation layout

### LINKED ISSUE #6550: Reconsider documentation website layout
Improve the current documentation website layout so that it is more palatable.
Previous links will still need to be reacheable.

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Implement a base documentation layout matching expected `lodestar` users.

Client side redirects are setup so that older links are not broken.

Fixes https://github.com/ChainSafe/lodestar/issues/6550

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -14,9 +14,10 @@
     "write-heading-ids": "docusaurus write-heading-ids"
   },
   "dependencies": {
-    "@docusaurus/core": "^3.2.0",
-    "@docusaurus/preset-classic": "^3.2.0",
-    "@docusaurus/theme-mermaid": "^3.2.0",
+    "@docusaurus/core": "^3.4.0",
```
Comment (nflaig) [inline docs/package.json:17]: do we need to update these as part of the PR?

Comment (jeluard) [reply docs/package.json:17]: As I added a new dependency (`@docusaurus/plugin-client-redirects`) defaulting to `3.4.0` and all docusaurus versions must match, I figured it made sense to just update all of them.

Comment (nflaig) [reply docs/package.json:17]: if it's related to PR changes we can keep it

**Review by philknows — APPROVED** [review]

File: docs/sidebars.ts [hunk 2]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
...
+          type: "category",
+          label: "Lodestar Light Client",
+          items: ["libraries/lightclient-prover/lightclient-cli", "libraries/lightclient-prover/lightclient"],
         },
         {
-          type: "doc",
-          label: "Prover",
-          id: "lightclient-prover/prover",
+          type: "category",
+          label: "Lodestar Light Prover",
```
Comment (nflaig) [inline docs/sidebars.ts:61]: "Lodestar Light Prover" is that the product name?

File: docs/sidebars.ts [hunk 3]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
...
+      label: "Developer Tools",
+      collapsed: false,
       items: [
-        "lightclient-prover/lightclient-cli",
         {
-          type: "doc",
-          label: "Light Client",
-          id: "lightclient-prover/lightclient",
+          type: "category",
+          label: "Lodestar Light Client",
```
Comment (nflaig) [inline docs/sidebars.ts:56]: we don't need to prefix this with Lodestar, it's redundant, and we don't do it for anything else

File: docs/pages/run/getting-started/installation.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Installation
 
+## Binaries
+
+Binaries can be downloaded from [this page](https://github.com/ChainSafe/lodestar/releases/) under the `Assets` section.
```
Comment (nflaig) [inline docs/pages/run/getting-started/installation.md:5]: You can always link to latest release directly

```
Binaries can be downloaded from [the release page](https://github.com/ChainSafe/lodestar/releases/latest) under the `Assets` section.
```

File: docs/pages/run/validator-management/vc-configuration.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Configuration
+title: Stake with a Validator Client
```
Comment (nflaig) [inline docs/pages/run/validator-management/vc-configuration.md:2]: hmm, again redundant to mentioned "Validator Client" here, and what else are you gonna do with a validator client besides staking?

File: docs/sidebars.ts [hunk 6]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
...
-        "beacon-management/beacon-cli",
-        "beacon-management/networking",
-        "beacon-management/mev-and-builder-integration",
-        "beacon-management/syncing",
-      ],
-    },
     {
       type: "category",
-      label: "Validator",
+      label: "Run A Node",
```
Comment (nflaig) [inline docs/sidebars.ts:8]: why is this collapsed? The content in "Run a node" is by far the most viewed, see https://github.com/ChainSafe/lodestar/issues/6550#issuecomment-2148635954. And it should be easy to access for users (and visible)

File: docs/sidebars.ts [hunk 7]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
...
       label: "Contributing",
+      collapsed: false,
       items: [
         {
-          type: "doc",
-          label: "Getting Started",
-          id: "contribution/getting-started",
+          type: "category",
+          label: "Advanced Topics",
+          collapsed: false,
```
Comment (nflaig) [inline docs/sidebars.ts:76]: please collapse as many more advanced topics as possible

File: docs/sidebars.ts [hunk 8]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
...
+          items: ["contribution/advanced-topics/setting-up-a-testnet"],
         },
         "contribution/depgraph",
         {
-          type: "doc",
-          label: "Dev CLI Reference",
-          id: "contribution/dev-cli",
```
Comment (nflaig) [inline docs/sidebars.ts:78]: dev cli seems gone as well (again)

File: docs/sidebars.ts [hunk 9]
```diff
@@ -3,83 +3,94 @@ import type {SidebarsConfig} from "@docusaurus/plugin-content-docs";
 const sidebars: SidebarsConfig = {
   tutorialSidebar: [
     "index",
-    "introduction",
-    {
-      type: "doc",
-      label: "Security",
-      id: "security",
-    },
```
Comment (nflaig) [inline docs/sidebars.ts:11]: I can't find deep references to those page anywhere, imo removing content should be separate from a "layout change" PR

File: docs/sidebars.ts (hunk 2 above)
Comment (jeluard) [reply docs/sidebars.ts:61]: typo

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.20.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.20.0) :tada:

**Pull request merged**